balancer-js/src/modules/zaps/bbausd2-migrations/bbausd1.integration.spec.ts

```diff
@@ -0,0 +1,256 @@
+    );
+    const _pool = await pools.findBy('address', poolAddress);
+    if (!_pool) throw new BalancerError(BalancerErrorCode.POOL_DOESNT_EXIST);
+    pool = _pool;
+  });
+
+  context('staked', async () => {
+    beforeEach(async function () {
+      this.timeout(20000);
+
+      // Stake them
+      await stake(signer, balance);
+    });
+
+    it('should transfer tokens from stable to boosted', async () => {
+      // Store balance before migration
+      const before = {
+        from: await getErc20Balance(gaugeAddress, signerAddress),
+        to: await getErc20Balance(addresses.bbausd2.gauge, signerAddress),
+      };
+
+      const query = migration.calldata(
+        before.from.toString(),
+        undefined,
+        signerAddress,
+        true,
+        authorisation,
+        pool.tokens.filter((token) => token.symbol !== 'bb-a-USD')
+      );
+
+      const { to, data } = query;
+      const gasLimit = MAX_GAS_LIMIT;
+      const response = await signer.sendTransaction({ to, data, gasLimit });
+
+      const reciept = await response.wait();
+      console.log('Gas used', reciept.gasUsed.toString());
+
+      const after = {
+        from: await getErc20Balance(gaugeAddress, signerAddress),
+        to: await getErc20Balance(addresses.bbausd2.gauge, signerAddress),
+      };
+
+      const diffs = {
+        from: after.from.sub(before.from),
+        to: after.to.sub(before.to),
+      };
+
+      console.log(diffs.from, diffs.to);
+
+      expect(diffs.from).to.eql(before.from.mul(-1));
+      expect(parseFloat(formatEther(diffs.to))).to.be.gt(0);
+    }).timeout(20000);
+  });
+
+  context('not staked', async () => {
+    it('should transfer tokens from stable to boosted', async () => {
+      // Store balance before migration
+      const before = {
+        from: await getErc20Balance(poolAddress, signerAddress),
+        to: await getErc20Balance(addresses.bbausd2.address, signerAddress),
+      };
+
+      const query = migration.calldata(
+        before.from.toString(),
+        undefined,
+        signerAddress,
+        false,
+        authorisation,
+        pool.tokens.filter((token) => token.symbol !== 'bb-a-USD')
+      );
+
+      const { to, data } = query;
+      const gasLimit = MAX_GAS_LIMIT;
+      const response = await signer.sendTransaction({ to, data, gasLimit });
+
+      const reciept = await response.wait();
+      console.log('Gas used', reciept.gasUsed.toString());
+
+      const after = {
+        from: await getErc20Balance(poolAddress, signerAddress),
+        to: await getErc20Balance(addresses.bbausd2.address, signerAddress),
+      };
+
+      const diffs = {
+        from: after.from.sub(before.from),
+        to: after.to.sub(before.to),
+      };
+
+      console.log(diffs.from, diffs.to);
+
+      expect(diffs.from).to.eql(before.from.mul(-1));
+      expect(parseFloat(formatEther(diffs.to))).to.be.gt(0);
+    }).timeout(20000);
+  });
+}).timeout(20000);
```

Comment:
Can we add a test that checks for revert when an artificialy high expectedAmount is passed.